src/WinRT.Runtime/Projections/ICustomPropertyProvider.net5.cs

```diff
@@ -23,6 +25,105 @@ interface ICustomProperty
+    /// </summary>
+#if EMBED
+    internal
+#else
+    public
+#endif
+    interface IBindableCustomPropertyImplementation
+    {
+        /// <summary>
+        /// Get the generated <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation representing the specified property name.
+        /// </summary>
+        /// <param name="name">The name of the property to get.</param>
+        /// <returns>The <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation for the property specified by <paramref name="name"/>.</returns>
+        public Microsoft.UI.Xaml.Data.BindableCustomProperty GetProperty(string name);
+
+        /// <summary>
+        /// Get the generated <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation representing the specified index property type.
+        /// </summary>
+        /// <param name="indexParameterType">The index property to get.</param>
+        /// <returns>The <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation for the property specified by <paramref name="indexParameterType"/>.</returns>
+        public Microsoft.UI.Xaml.Data.BindableCustomProperty GetProperty(Type indexParameterType);
+    }
+
+    /// <summary>
+    /// An <see cref="Microsoft.UI.Xaml.Data.ICustomProperty"/> implementation that relies on a source generation approach for its implememtation
+    /// rather than reflection.  This is used by the source generator generating the implementation for <see cref="Microsoft.UI.Xaml.Data.IBindableCustomPropertyImplementation"/>.
+    /// </summary>
+    [global::WinRT.WinRTExposedType(typeof(global::ABI.Microsoft.UI.Xaml.Data.ManagedCustomPropertyWinRTTypeDetails))]
+#if EMBED
+    internal
+#else
+    public
+#endif
+    sealed class BindableCustomProperty : ICustomProperty
+    {
+        private readonly bool _canRead;
+        private readonly bool _canWrite;
+        private readonly string _name;
+        private readonly Type _type;
+        private readonly Func<object, object> _getValue;
+        private readonly Action<object, object> _setValue;
+        private readonly Func<object, object, object> _getIndexedValue;
+        private readonly Action<object, object, object> _setIndexedValue;
+
+        public BindableCustomProperty(
+            bool canRead,
+            bool canWrite,
+            string name,
+            Type type,
+            Func<object, object> getValue,
+            Action<object, object> setValue,
+            Func<object, object, object> getIndexedValue,
+            Action<object, object, object> setIndexedValue)
+        {
+            _canRead = canRead;
+            _canWrite = canWrite;
+            _name = name;
+            _type = type;
+            _getValue = getValue;
+            _setValue = setValue;
+            _getIndexedValue = getIndexedValue;
+            _setIndexedValue = setIndexedValue;
+        }
+
+        bool ICustomProperty.CanRead => _canRead;

```

Comment:
How come all members are explicitly implemented?

Reply:
ICustomProperty is internal and is just an internal implementation detail.  So explicitly implemented them.

Reply:
Oh, duh, right ahahah